src/getinfo.sml

```diff
@@ -163,32 +163,31 @@ fun findFirstExpAfter env str fileName {line = line, char = char} =
 
 
 datatype foundInEnv = FoundStr of (string * Elab.sgn)
-                    | FoundCon of (string * Elab.kind)
-                    | FoundExp of (string * Elab.con)
+                    | FoundKind of (string * Elab.kind)
+                    | FoundCon of (string * Elab.con)
 
 fun getNameOfFoundInEnv (f: foundInEnv) =
     case f of
         FoundStr (x, _) => x
+      | FoundKind (x, _) => x
       | FoundCon (x, _) => x
-      | FoundExp (x, _) => x
 
 fun filterSgiItems (items: Elab.sgn_item list) : foundInEnv list =
     let
         fun mapF item =
             case item of
-                (Elab.SgiVal (name, _, c), _) => [FoundExp (name, c)]
-              | (Elab.SgiCon (name, _, k, _), _) => [FoundCon (name, k)]
+                (Elab.SgiVal (name, _, c), _) => [FoundCon (name, c)]
+              | (Elab.SgiCon (name, _, k, _), _) => [FoundKind (name, k)]
               | (Elab.SgiDatatype ds, loc) =>
                 List.concat (List.map (fn (dtx, i, _, cs) =>
-                                          FoundExp (dtx, (Elab.CNamed i, loc))
-                                          :: List.map (fn (x, i, _) => FoundExp (x, (Elab.CRel i, loc))) cs) ds)
+                                          FoundCon (dtx, (Elab.CNamed i, loc))
```

Comment:
My last comment here must have been unclear, or I'm stuck on some misunderstanding.  A `datatype` declaration introduces both a type-level identifier (the datatype itself) and, in general, several value-level identifiers (the constructors).  It seems suspicious that the datatype and its constructors are being declared with the same constructor, whether it's called `FoundCon` or `FoundExp`.  Are you sure that's right?

Reply:
I totally misunderstood your comment and failed to see my fault. Sorry! Fixing it now.